src/Application/FileProcessor/PhpFileProcessor.php

```diff
@@ -80,11 +84,20 @@ public function process(File $file, Configuration $configuration): array
             // important to detect if file has changed
             $this->printFile($file, $configuration);
 
-            if ($file->hasChanged()) {
+            $fileHasChangedInCurrentPass = $file->hasChanged();
+
+            if ($fileHasChangedInCurrentPass) {
```

Comment:
immediate assign can be used:

```suggestion
            if ($fileHasChangedInCurrentPass) {
                $fileHasChanged = true;
```